pkg/suggestion/v1alpha2/bayesianoptimization/src/algorithm_manager.py

```diff
@@ -0,0 +1,235 @@
+
+
+def deal_with_discrete(feasible_values, current_value):
+    """ function to embed the current values to the feasible discrete space"""
+    diff = np.subtract(feasible_values, current_value)
+    diff = np.absolute(diff)
+    return feasible_values[np.argmin(diff)]
+
+
+def deal_with_categorical(feasible_values, one_hot_values):
+    """ function to do the one hot encoding of the categorical values """
+    index = np.argmax(one_hot_values)
+    #index = one_hot_values.argmax()
+    return feasible_values[int(index)]
+
+
+class AlgorithmManager:
+    """ class for the algorithm manager
+    provide some helper functions
+    """
+
+    def __init__(self, experiment_name, experiment, X_train, y_train, logger=None):
+        self.logger = logger if (logger is not None) else get_logger()
+        self._experiment_name = experiment_name
+        self._experiment = experiment
+        self._goal = self._experiment.spec.objective.type
+        self._dim = 0
+        self._lowerbound = []
+        self._upperbound = []
+        self._types = []
+        self._names = []
+        # record all the feasible values of discrete type variables
+        self._discrete_info = []
+        self._categorical_info = []
+        self._name_id = {}
+
+        self._parse_config()
+
+        self._X_train = self._mapping_params(X_train)
+        self.parse_X()
+
+        self._y_train = y_train
+        self._parse_metric()
+
+    @property
+    def experiment_name(self):
+        """ return the study id """
+        return self._experiment_name
+
+    @property
+    def experiment(self):
+        """ return the study configuration """
+        return self._experiment
+
+    @property
+    def goal(self):
+        """ return the optimization goal"""
+        return self._goal
+
+    @property
+    def dim(self):
+        """ return the dimension """
+        return self._dim
+
+    @property
+    def lower_bound(self):
+        """ return the lower bound of all the parameters """
+        return self._lowerbound
+
+    @property
+    def upper_bound(self):
+        """ return the upper bound of all the parameters """
+        return self._upperbound
+
+    @property
+    def types(self):
+        """ return the types of all the parameters """
+        return self._types
+
+    @property
+    def names(self):
+        """ return the names of all the parameters """
+        return self._names
+
+    @property
+    def discrete_info(self):
+        """ return the info of all the discrete parameters """
+        return self._discrete_info
+
+    @property
+    def categorical_info(self):
+        """ return the info of all the categorical parameters """
+        return self._categorical_info
+
+    @property
+    def X_train(self):
+        """ return the training data """
+        return self._X_train
+
+    @property
+    def y_train(self):
+        """ return the target of the training data"""
+        return self._y_train
+
+    def _parse_config(self):
```

Comment:
These utilities are already added in https://github.com/kubeflow/katib/blob/master/pkg/suggestion/v1alpha2/parsing_util.py  See https://github.com/kubeflow/katib/pull/435/files